rethinkdb/datadog_checks/rethinkdb/document_db/query.py

```diff
@@ -0,0 +1,118 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from typing import Any, Callable, Iterator, List, Mapping, Sequence, Tuple
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.base.log import CheckLoggingAdapter
+
+from .types import Enumeration, Group, Metric, MetricSpec, Modifier
+from .utils import dotted_join, lookup_dotted, to_time_elapsed
+
+
+class DocumentQuery(object):
+    """
+    A helper for retrieving metrics from document-oriented ("JSON") databases.
+    """
```

Comment:
Could add some docs about how this query engine works. And what are each type of queries (metrics, enumerations, groups) and probably how they work ?

Reply:
I figured out the tests would serve as sufficient documentation… Don't want to duplicate info as it's very likely to become stale at some point. I can add a mention to look at the tests for usage though.

Reply:
Actually docs are a good idea, esp for when this ends up being added to the docs site. :-)